joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -58,9 +59,25 @@ trait AstCreatorHelper {
     }
   }
 
-  private def fileLines(node: IASTNode): Seq[Int] = {
+  private def fileOffsetTable(node: IASTNode): Array[Int] = {
     val f = fileName(node)
-    global.file2LinesCache.computeIfAbsent(f, _ => IOUtils.readLineLengthsInFile(Paths.get(f)))
+    global.file2OffsetTable.computeIfAbsent(f, _ => genFileOffsetTable(Paths.get(f)))
+  }
+
+  private def genFileOffsetTable(fileName: Path): Array[Int] = {
+    val bytes    = Files.readAllBytes(fileName)
+    var asString = new String(bytes, StandardCharsets.UTF_8)
+    asString = asString.replaceAll("\r\n", "\n")
+    asString = asString.replaceAll("\r", "\n")
+    val asCharArray = asString.toCharArray
+    val offsets     = mutable.ArrayBuffer.empty[Int]
+
+    for (i <- Range(0, asCharArray.length)) {
+      if (asCharArray(i) == '\n') {
+        offsets.append(i + 1)
+      }
+    }
```

Comment:
Is this truly platform-independent? The IOUtils file reading from cpg does quite some stuff to handle all kinds of problems with charset encoding and unmappable characters.

Reply:
I changed the file reading to use IOUtils.